src/lint/colors/tests/astro-color-is-using-correct-theme.ts

```diff
@@ -0,0 +1,66 @@
+import { AstroTheme } from "../../../types/tokens";
+import { FillStyleNode } from "../../../types/figma";
+import { LintingResult } from "../../../types/results";
+import { stripToLoadableId } from "../../../tokens";
+import { tokens } from "../../../tokens";
+const { colorTokens } = tokens();
+
+const astroColorIsUsingCorrectTheme = (
+  node: FillStyleNode,
+  theme: AstroTheme
+): Promise<LintingResult> => {
+  return new Promise((resolve) => {
+    const test = "Using Astro Color in Correct Theme";
+    const name = node.name;
+    const fillStyleId = (node.fillStyleId as string) || "";
+    const astroColor = colorTokens.get(stripToLoadableId(fillStyleId));
+    const fills = node.fills;
+    let pass = false;
+    let message = "";
+    if (astroColor?.name) {
+      const astroColorNameWithTheme = `${theme}/${astroColor?.name}`;
+      const astroColorWithTheme = colorTokens.get(astroColorNameWithTheme);
+      pass = astroColor?.id === astroColorWithTheme?.id ? true : false;
+      message = pass
+        ? `Node is using a fill style (${astroColor.name}) from Astro but it's not the correct theme (${theme})`
+        : `Node is using a fill style (${astroColor.name}) from Astro in the correct theme (${theme})`;
+    } else if (Array.isArray(fills) && fills.length === 0) {
```

Comment:
If you can, I would refactor this chain of else ifs to a switch for better readability and logic flow.